core/client/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -325,4 +325,11 @@ public void unmount(AlluxioURI path, UnmountOptions options)
       mFileSystemContext.releaseMasterClient(masterClient);
     }
   }
+
+  @Override
+  public void close() {
+    if (mFileSystemContext != FileSystemContext.INSTANCE) {
+      mFileSystemContext.close();
```

Comment:
good point. I've removed FileSystem#Close and leave the user to explicitly manage the lifetime of the context. 

So in AbstractFileSystem, I am closing the context there.